lightning/src/ln/channelmanager.rs

```diff
@@ -4134,7 +4170,7 @@ where
 												}
 											}
 										}
-										if let PendingHTLCRouting::Forward { onion_packet, .. } = routing {
+										if let PendingHTLCRouting::Forward { ref onion_packet, .. } = routing {
```

Comment:
We will always fail to decode `onion_packet` if it's a phantom, because we use `Recipient::Node` instead of `::PhantomNode` when decoding the encrypted TLVs. Not much we can do here until we add full phantom support, IIUC.